### PR TITLE
Enhanced useBraces modes, structural indentation, and nextControlFlowPosition fixes

### PR DESCRIPTION
**Pull request by steffen-heil-secforge**
This fork enhances formatting control and fixes several bugs in control flow positioning.

### Enhanced useBraces Modes
- **whenFormattedMultiLine**: Only adds braces when the body would be formatted across multiple lines
- **onlyNeeded**: Only adds braces when syntactically required (stricter than whenNeeded)

These modes provide finer control over brace usage in control flow statements.

### Structural Indentation for Ternary Operators
- **conditionalExpressionIndentStyle: structural**: Prevents deep nesting in ternary operators by using a flatter indentation structure
- Particularly useful for complex nested ternary expressions

### Expression Parentheses Control
- **forceParentheses** option: Controls when wrapping parentheses are added around expressions
- Helps avoid unnecessary parentheses while maintaining clarity

## Bug Fixes

- Fixed NextLineExceptAfterBrace causing infinite loops with try-catch-finally statements
- Fixed NextLineExceptAfterBrace for do-while statements to properly detect closing braces
- Fixed useBraces:whenNeeded incorrectly removing braces from function and variable declarations (required in strict mode)
- Fixed multi-statement blocks incorrectly being treated as single statements

## Performance Improvements

- Eliminated heap allocations in brace-checking helpers by using iterator short-circuiting
- Conditions are now only created when NextLineExceptAfterBrace is actually configured
- Extracted common patterns into reusable helper functions

## Configuration

- Version: 0.95.11+sf.2
- Plugin URLs are now configurable via Cargo.toml metadata
- Dependencies use version ranges instead of exact pinning for better compatibility

## Test Coverage

- Added 25 new test files covering edge cases in brace usage, empty statements, and control flow positioning
- Added regression tests for empty blocks and NextLineExceptAfterBrace behavior
- All 671 tests passing

### CODE REVIEW

File: src/configuration/types.rs [hunk 1]
```diff
@@ -163,14 +176,20 @@ pub enum UseBraces {
...
 }
 
 generate_str_to_from![
   UseBraces,
   [Maintain, "maintain"],
   [WhenNotSingleLine, "whenNotSingleLine"],
   [Always, "always"],
-  [PreferNone, "preferNone"]
+  [PreferNone, "preferNone"],
+  [OnlyNeeded, "onlyNeeded"],
```
Comment (yassir-kozha-secforge) [inline src/configuration/types.rs]:
```suggestion
  [WhenNeeded, "whenNeeded"],
```

Comment (steffen-heil-secforge) [reply src/configuration/types.rs]: done

**Comment by steffen-heil-secforge** [comment]
@claude Check this

File: debug_test.js [hunk 2]
```diff
@@ -0,0 +1 @@
+if (true) { a; } else { b; }
```
Comment (yassir-kozha-secforge) [inline debug_test.js]: ?

**Comment by steffen-heil-secforge** [comment]
@claude check this

File: debug_test.js (hunk 2 above)
Comment (steffen-heil-secforge) [reply debug_test.js]: this should never have been committed. and it was removed now.
